ase/sw/app_backend.c

```diff
@@ -1340,56 +1342,76 @@ bool deallocate_buffer_by_index(int search_index)
 {
     FUNC_CALL_ENTRY;
     bool value;
-    uint64_t *bufptr = (uint64_t *) NULL;
-    struct wsmeta_t *wsptr;
+	struct buffer_t *bufptr = (struct buffer_t *) NULL;
```

Comment:
'wsid' may not necessarily be buffer_t->index. Is this difference being accounted for?

For eg.
**In MCP mode,** 
OPAE API buffer wsid=0 maps to ASE API Buffer 2 (this offset is due to MMIO and UMAS regions)
**In DCP mode,** 
OPAE API buffer wsid=0 = ASE API Buffer 1 (this offset is due to MMIO region)

This can cause the wrong region to get deallocated.

Reply:
`wsptr->index` contains the same value as `buf->index` (see l. 1237 in the original), so shouldn't a search for either work?

Reply:
In allocate_buffer() asebuf_index_count keeps track of all buffers allocated in the application. ASE internally counts MMIO and UMSG regions as 0 and 1 respectively. App index starts from 1 or 2 depending on platform type.

Reply:
From the original code, there is no difference regarding index between ws->index and buf->index. Please check the line at 1237 of app_backend.c. No matter how it is interpreted on different platforms, the index is same in wsmeta and buffer as shown in line 1237.. So it is not necessary to create another data structure of wsmeta_t to keep track of the index.

Reply:
@rahulrs Please provide code evidence for "'wsid' may not necessarily be buffer_t->index". If evidence is not provided by Sunday (11/26) and DCP tests pass, we will merge this pull request branch into master branch